debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -279,6 +281,26 @@ public void shouldReceiveChangesForDeletes() throws Exception {
         assertEquals(topicName, record.topic());
         VerifyRecord.isValidTombstone(record, PK_FIELD, 2);
     }
+
+    @Test
+    public void shouldReceiveNumericTypeAsDouble() throws Exception {
+        PostgresConnectorConfig config = new PostgresConnectorConfig(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, PostgresConnectorConfig.DecimalHandlingMode.DOUBLE)
+                .build());
+        PostgresTaskContext context = new PostgresTaskContext(config, new PostgresSchema(config));
+        recordsProducer = new RecordsStreamProducer(context, new SourceInfo(config.serverName()));
+
+        TestHelper.executeDDL("postgres_create_tables.ddl");
+
+        consumer = testConsumer(1);
+        recordsProducer.start(consumer);
+
+        List<SchemaAndValueField> schemasAndValuesForNumericType = schemasAndValuesForNumericType();
+        schemasAndValuesForNumericType.set(3, new SchemaAndValueField("d", Schema.OPTIONAL_FLOAT64_SCHEMA, 1.1d));
+        schemasAndValuesForNumericType.set(4, new SchemaAndValueField("n", Schema.OPTIONAL_FLOAT64_SCHEMA, 22.22d));
+
+        assertInsert(INSERT_NUMERIC_TYPES_STMT, schemasAndValuesForNumericType);
```

Comment:
Got it. I've rebased your change and adapted it slightly, it's `PostgresValueConverter#converter()` now which must return the correct conversion method based on the decimal handling mode. With that, the test passes.